jumanji/environments/routing/rware/env_test.py

```diff
@@ -0,0 +1,215 @@
+    # Check that discounts have the correct number of dimensions
+    assert jnp.ndim(timestep1.discount) == 0
+    assert jnp.ndim(timestep.discount) == 0
+    # Check that the state is made of DeviceArrays, this is false for the non-jitted
+    # step function since unpacking random.split returns numpy arrays and not device arrays.
+    assert_is_jax_array_tree(new_state1)
+    # Check that the state has changed
+    assert new_state1.step_count != state.step_count
+    assert not jnp.all(new_state1.grid != state.grid)
+    # Check that two different actions lead to two different states
+    new_state2, timestep2 = step_fn(state, action2)
+    assert not jnp.all(new_state1.grid != new_state2.grid)
+
+    jax.debug.print("grid: {g}", g=state.grid)
+
+    # Check that the state update and timestep creation work as expected
+    agents = state.agents
+    agent = tree_slice(agents, 1)
+    jax.debug.print("agents: {g}", g=agents)
+    x = agent.position.x
+    y = agent.position.y
+
+    # turning and moving actions
+    actions = [2, 2, 3, 3, 1, 3, 1]
+
+    # Note: starting direction is 3 (facing left)
+    new_locs = [
+        (x, y, 2),  # turn left -> facing down
+        (x, y, 1),  # turn left -> facing right
+        (x, y, 2),  # turn right -> facing down
+        (x, y, 3),  # turn right -> face left
+        (x, y - 1, 3),  # move forward -> move left
+        (x, y - 1, 0),  # turn right -> face up
+        (x - 1, y - 1, 0),  # move forward -> move up
+    ]
+
+    for action, new_loc in zip(actions, new_locs):
+        state, timestep = step_fn(state, jnp.array([action, action]))
+        agent1_info = tree_slice(state.agents, 1)
+        agent1_loc = (
+            agent1_info.position.x,
+            agent1_info.position.y,
+            agent1_info.direction,
+        )
+        assert agent1_loc == new_loc
+
+
+def test_rware__does_not_smoke(rware_env: Rware) -> None:
+    """Validate that we can run an episode without any errors."""
+    check_env_does_not_smoke(rware_env)
+
+
+def test_rware__time_limit(rware_env: Rware) -> None:
+    """Validate the terminal reward."""
+    step_fn = jax.jit(rware_env.step)
+    state_key = random.PRNGKey(10)
+    state, timestep = rware_env.reset(state_key)
+
+    for _ in range(rware_env.time_limit - 1):
+        state, timestep = step_fn(state, jnp.array([0, 0]))
+
+    assert timestep.mid()
+    state, timestep = step_fn(state, jnp.array([0, 0]))
+    assert timestep.last()
+
+
+def test_rware__truncation(
+    deterministic_rware_env: Tuple[Rware, State, TimeStep]
+) -> None:
+    """Validate episode truncation based on set time limit."""
+    rware_env, state, timestep = deterministic_rware_env
+    step_fn = jax.jit(rware_env.step)
+
+    # truncation
+    for _ in range(rware_env.time_limit):
+        state, timestep = step_fn(state, jnp.array([0, 0]))
+
+    assert timestep.last()
+    assert not jnp.all(timestep.discount == 0)
+
+
+def test_rware__terminate_upon_collision(
```

Comment:
Correct me if I'm wrong, but seems like you're truncating upon collision because you're expecting that not all discounts are 0.

```suggestion
def test_rware__truncate_upon_collision(
```